apps/aehttp/test/aehttp_contracts_SUITE.erl

```diff
@@ -544,42 +651,200 @@ maps_contract(Config) ->
     _DBal0 = ensure_balance(DPubkey, 500000),
     {ok,[_]} = aecore_suite_utils:mine_key_blocks(NodeName, 1),
 
-    %% Compile contract "maps.aes" but a simple test first.
-    ContractString = aeso_test_utils:read_contract("maps"),
-    BinCode = aeso_compiler:from_string(ContractString, []),
-    HexCode = aeu_hex:hexstring_encode(BinCode),
+    %% Compile test contract "maps.aes".
+    MCode = compile_test_contract("maps"),
 
     %% Initialise contract owned by Alice.
-    {EncodedContractPubkey,_,_} =
-	create_contract(NodeName, APubkey, APrivkey, HexCode, <<"()">>),
+    {EncodedMapsPubkey,DecodedMapsPubkey,_} =
+       create_compute_contract(NodeName, APubkey, APrivkey, MCode, <<"()">>),
 
-    %% get_state
-    ct:pal("Calling get_state\n"),
-    GetState1 = call_get_state(NodeName, BPubkey, BPrivkey,
-			       EncodedContractPubkey, HexCode),
-    ct:pal("Get state 1 ~p\n", [GetState1]),
+    %% Some functions can't be called directly, we need an interface contract.
```

Comment:
Is this:
* Because of feature not implement yet. In this case Pivotal Tracker ticket needed. Or
* By design. Or
* Because of ease of testing. In this case a comment here would help.

Reply:
There is no way to create a record as an argument